src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -98,22 +98,13 @@ internal sealed class ContainedDocument : ForegroundThreadAffinitizedObject, IVi
             _optionService = _workspace.Services.GetService<IOptionService>();
             _hostType = GetHostType();
 
-            var rdt = (IVsRunningDocumentTable)componentModel.GetService<SVsServiceProvider>().GetService(typeof(SVsRunningDocumentTable));
-
-            IVsHierarchy sharedHierarchy;
-            uint itemIdInSharedHierarchy;
-            var isSharedHierarchy = LinkedFileUtilities.TryGetSharedHierarchyAndItemId(hierarchy, itemId, out sharedHierarchy, out itemIdInSharedHierarchy);
-
-            var filePath = isSharedHierarchy
-                ? rdt.GetMonikerForHierarchyAndItemId(sharedHierarchy, itemIdInSharedHierarchy)
-                : rdt.GetMonikerForHierarchyAndItemId(hierarchy, itemId);
-
-            // we couldn't look up the document moniker in RDT for a hierarchy/item pair
-            // Since we only use this moniker as a key, we could fall back to something else, like the document name.
-            if (filePath == null)
+            string filePath;
+            if (!ErrorHandler.Succeeded(((IVsProject)hierarchy).GetMkDocument(itemId, out filePath)))
```

Comment:
We didn't previously depend on the hierarchy being an IVsProject - are there any cases where this might fail? For example, I believe the stub hierarchies provided during asynchronous solution load are not guaranteed to implement IVsProject. I don't think we'd be constructing a ContainedDocument within one of those stub hierarchies, but are there other ways to get non-IVsProject hierarchies here?